### DISCUSSION TITLE
First day of the week in the iOS Contribution Graph widget

### DISCUSSION THREAD

**Discussion by FrankKooij** [community/community#3374] [Mobile] [answered]
The iOS Contribution Graph widget shows Sunday as the first day of the week. For Europeans, however, the week starts on Monday. Would it be possible to enable the user to choose which day of the week should be represented by the top row of the Contribution Graph widget?

**Comment by candyho** [comment, marked as answer]
Hey, thanks for your feedback. We hear you! We had a brief look at this issue and unfortunately the effort we need to put in might not be trivial and we can't afford to prioritise this change at the moment. We will keep monitoring related feedback and re-evaluate the priority from time to time. A different start day in Europe is something we'll definitely keep in mind when we develop new features in the near future. Thanks again for flagging this and sorry for the inconvenience caused!

**Reply by rjc** [reply]
Just a quick note that it isn't just Europe - it's international, as in ISO - the organisation - not just ISO 8601. If anything US, with Sunday as the first day of the week, is in minority. It would be great if you could change the default to be international (ISO) first day of the week - Monday. If not, then please make it configurable at the very least. GitLab somehow managed to do it... years ago ;^)

**Comment by ownmas** [comment]
I want this feature too!

@candyho, it's actually super-easy to implement – https://developer.apple.com/documentation/foundation/calendar/2293656-firstweekday

**Reply by candyho** [reply]
Hey, thanks for digging into this. Checked in with engineer, while we agree it might be easy to write some code to use firstWeekday, we realised there're other technical implications such as reformatting all of the data that's returned by the API  which would require bigger effort to resolve. We don't really have an easy and straightforward solution at this point. Sorry about that and thanks for your understanding.

**Reply by jmwielandt** [reply]
whats about just a patch on the frontend? until you figure out how to propagate that configuration through the backend and all the apis.
Also @rjc was right about the standard ISO (https://github.com/github-community/community/discussions/3374#discussioncomment-2196867): https://www.iso.org/iso-8601-date-and-time-format.html


Check this out:
https://www.unicode.org/cldr/cldr-aux/charts/30/supplemental/territory_information.html

Here, @basicallydan pointed out that this topic has also to do with religion and traditions, not only a per-country setting. https://github.com/isaacs/github/issues/644#issuecomment-208804094
